### ISSUE TITLE
Guru Meditation Error - EZ-FLASH Parallel

### ISSUE THREAD

**Issue by deneraraujo** [DS-Homebrew/GodMode9i#241]
Guru Meditation Error (pic attached).
Flashcart: EZ-FLASH Parallel.
System: NDS Lite.
![20240607_183122](https://github.com/DS-Homebrew/GodMode9i/assets/2524788/58706909-b987-4286-b639-3603bd8925dc)

**Comment by mkeseattle** [comment]
The error also occurs on other systems such as the Nintendo 3DS LL and the Nintendo DSi LL.

EZ-FLASH Parallel with Kernel 1.06, GodModei9 v3.5.4

https://www.ezflash.cn/download/

**Comment by NubbbZ** [comment]
can confirm that it happens for me as well, DS Lite, EZ-Flash Parallel with Kernel 1.06 and GodModei9 v3.6.0

**Comment by Epicpkmn11** [comment]
Is this fixed by @lifehackerhansol's new driver? (used by TWiLight Menu++ [v27.12.2](https://github.com/DS-Homebrew/TWiLightMenu/releases/tag/v27.12.2))

**Comment by keyofnight** [comment]
Same issue here with EZ-Flash Parallel with Kernel 1.06 and Godmodei9 v3.7.0.

**Comment by mkeseattle** [comment]
> Is this fixed by [@lifehackerhansol](https://github.com/lifehackerhansol)'s new driver? (used by TWiLight Menu++ [v27.12.2](https://github.com/DS-Homebrew/TWiLightMenu/releases/tag/v27.12.2))

Can confirm the new driver works. TWiLight Menu++ v27.12.2 up to v27.13.0 it is fixed.

@lifehackerhansol
@Epicpkmn11

**Comment by mkeseattle** [comment]
> > Is this fixed by [@lifehackerhansol](https://github.com/lifehackerhansol)'s new driver? (used by TWiLight Menu++ [v27.12.2](https://github.com/DS-Homebrew/TWiLightMenu/releases/tag/v27.12.2))
> 
> Can confirm the new driver works. TWiLight Menu++ v27.12.2 up to v27.13.0 it is fixed.
> 
> [@lifehackerhansol](https://github.com/lifehackerhansol) [@Epicpkmn11](https://github.com/Epicpkmn11)

@RocketRobz 

Can you reopen? Because the file only works in combination with TWiLight Menu++.

- Autoboot -> EZ Flash Parallel -> ezds.dat

If uses this file only with EZ FLASH Parallel with Kernel 1.06 comes a message at startup this message /_nds/TWiLightMenu/main.srldr not found.
Press B to return to menu.

**Comment by RocketRobz** [comment]
> Can you reopen? Because the file only works in combination with TWiLight Menu++.
> 
> * Autoboot -> EZ Flash Parallel -> ezds.dat
> 
> If uses this file only with EZ FLASH Parallel with Kernel 1.06 comes a message at startup this message /_nds/TWiLightMenu/main.srldr not found. Press B to return to menu.

wdym? Does that message appear even with `main.srldr` already in `fat:/_nds/TWiLightMenu/`?

**Comment by mkeseattle** [comment]
> > Can you reopen? Because the file only works in combination with TWiLight Menu++.
> > 
> > * Autoboot -> EZ Flash Parallel -> ezds.dat
> > 
> > If uses this file only with EZ FLASH Parallel with Kernel 1.06 comes a message at startup this message /_nds/TWiLightMenu/main.srldr not found. Press B to return to menu.
> 
> wdym? Does that message appear even with `main.srldr` already in `fat:/_nds/TWiLightMenu/`?

If only use the kernel 1.06 for EZ FLASH Parallel Godmode9i does not work. The kernel 1.06 is based on some oldest Wood kernel version from R4.

If do not use EZP kernel and only TWiLightMenu++ with the corresponding file from Autoboot folder and copy everything from _nds to SD card and start TWiLightMenu++ and open Godmode9i it works without this error message and can use it normally.

Would like Godmode9i to also work with EZP Kernel. Is this possible?

**Comment by mkeseattle** [comment]
> > If only use the kernel 1.06 for EZ FLASH Parallel Godmode9i does not work. The kernel 1.06 is based on some oldest Wood kernel version from R4.
> > If do not use EZP kernel and only TWiLightMenu++ with the corresponding file from Autoboot folder and copy everything from _nds to SD card and start TWiLightMenu++ and open Godmode9i it works without this error message and can use it normally.
> > Would like Godmode9i to also work with EZP Kernel. Is this possible?
> 
> Not without manually DLDI-patching the `.nds` file using the fixed DLDI driver from TWLMenu++, and then disabling auto-patching support in the `.nds` file. Here's the fixed driver: [ez5n.zip](https://github.com/user-attachments/files/19769832/ez5n.zip)
> 
> Use dlditool to DLDI-patch, and Dldino-apatch to disable auto-patching.

Ok thanks will try later.

What are the current version of dlditool and dldino-apatch? Is there an All in One tool for this task on GitHub?

**Comment by lifehackerhansol** [comment]
You can replace `_ezds/EZDS_DLDI.dldi` with the file in the zip Robz posted above. Make sure the names match.

It should theoretically work. I haven't tried this.

**Comment by mkeseattle** [comment]
> You can replace `_ezds/EZDS_DLDI.dldi` with the file in the zip Robz posted above. Make sure the names match.
> 
> It should theoretically work. I haven't tried this.

Not working. Another Guru Meditation Error

![Image](https://github.com/user-attachments/assets/49a00859-1f8b-469d-932a-7f6902948409)

**Comment by RocketRobz** [comment]
> Not working. Another Guru Meditation Error
> 
> ![Image](https://github.com/user-attachments/assets/49a00859-1f8b-469d-932a-7f6902948409)

Then download dlditool and dldi-noap from here: https://www.chishm.com/DLDI/#tools
Download both `Win32 GUI` and `No auto-patch GUI`.

**Comment by mkeseattle** [comment]
> > Not working. Another Guru Meditation Error
> > ![Image](https://github.com/user-attachments/assets/49a00859-1f8b-469d-932a-7f6902948409)
> 
> Then download dlditool and dldi-noap from here: https://www.chishm.com/DLDI/#tools Download both `Win32 GUI` and `No auto-patch GUI`.

Not working.

**Comment by RocketRobz** [comment]
> Not working.

What did you do, exactly?

**Comment by mkeseattle** [comment]
> > Not working.
> 
> What did you do, exactly?

There is another Guru Mediation Error that is almost similar.

I just patched the `Godmode9i.nds v3.7.0` file with the first tool and the second and only with the first tool.

First + second

![Image](https://github.com/user-attachments/assets/91959276-3ee8-4874-ab59-8a403d7a1687)

First only

![Image](https://github.com/user-attachments/assets/1f42abce-de46-4d51-91be-346d81f808ad)

**Comment by RocketRobz** [comment]
> I just patched the `Godmode9i.nds v3.7.0` file with the first tool and the second and only with the first tool.

For the second tool, did you run `DLDI No-Patch.exe` or `DLDI Yes-Patch.exe`?
Only `DLDI No-Patch.exe` needs to be run.

If you already did, then you might be out of luck for running GodMode9**i** with the EZP kernel.

**Comment by mkeseattle** [comment]
> > I just patched the `Godmode9i.nds v3.7.0` file with the first tool and the second and only with the first tool.
> 
> For the second tool, did you run `DLDI No-Patch.exe` or `DLDI Yes-Patch.exe`? Only `DLDI No-Patch.exe` needs to be run.
> 
> If you already did, then you might be out of luck for running GodMode9**i** with the EZP kernel.

I only used `DLDI No-Patch.exe. What exactly is the problem? An EZP kernel problem so this flash card can't run it?

**Comment by RocketRobz** [comment]
> I only used `DLDI No-Patch.exe` What exactly is the problem? An EZP kernel problem so this flash card can't run it?

Yes, it is an EZP kernel problem if other loaders such as TWiLight Menu++ (and probably nds-hb-menu as well) can run it just fine.
